### PR TITLE
Updated markdown item ranges

### LINKED ISSUE #1107: Properly determine markdown item insertion delimiter
Should inspect the leading item marker to see if they're using `- `, `* `, etc to delimit their lists.  Don't forget the newline

These seem to behave a bit like indentation.  The item content should actually go between the leading delimiter and the newline, just like with indentation.  So I don't think we can just treat it like a simple insertion delimiter.  We probably should try to find a way to reuse the pieces of `editNew` that we use for line targets, rather than simple delimiter insertion

A few tests we'll want:

- [ ] `"bring item air before bat"`
- [ ] `"bring item air after bat"`
- [ ] `"drink item air"`

### LINKED ISSUE #1199: Leading indentation of next item included in `"item"` content range
In the following markdown:

```markdown
- aaa
  - bbb
```

The scope `"item air"` includes from `aaa` until the `-` on the next line. For example, saying `"round wrap item air"` results in the following:

```markdown
- (aaa
  )- bbb
```

### LINKED ISSUE #1200: `"bring item"` broken for indented lists
In the following markdown:

```markdown
- aaa
  - bbb
  - ccc
  - ddd
  - eee
```

Saying `"bring item each after bat"` results in the following:

```markdown
- aaa
  - bbb
  - ccc
    eee
  - ddd
  - eee
```

It seems like this bug is in part because the leading indentation of `ccc` is included in item `- bbb` (#1199), but in addition, it seems that the range is being extended to the end of the line.

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
Fixes #1200
Fixes #1199
Fixes #1107

## Checklist

- [x] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [ ] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [ ] I have not broken the cheatsheet

### CODE REVIEW

File: packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/languages/markdown/chuckItem2.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      - bbb
+      - ddd
+  selections:
+    - anchor: {line: 1, character: 5}
+      active: {line: 1, character: 5}
+  marks: {}
+finalState:
+  documentContents: |-
+    - aaa
+
```
Comment (pokey) [inline packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/languages/markdown/chuckItem2.yml:24]: so is the idea to handle this one as part of [#855](https://github.com/cursorless-dev/cursorless/issues/855)?

**Review by pokey — APPROVED** [review]
Looks good with minor comment / question

File: packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/languages/markdown/chuckItem2.yml (hunk 1 above)
Comment (AndreasArvidsson) [reply packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/languages/markdown/chuckItem2.yml:24]: Maybe? For now this is just an a improvement of what we had, but not perfect.

Comment (pokey) [reply packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/languages/markdown/chuckItem2.yml:24]: Seems fine

**Pull request merged**